agent/pkg/podman/podman.go

```diff
@@ -62,10 +60,13 @@ type PodmanClient struct {
 	mu         sync.Mutex
 	wg         waitgroupx.Group
 	containers map[cproto.ID]*PodmanContainer
+	agentTmp   string
 }
 
 // New returns a new podman client, which launches and tracks containers.
-func New(opts options.PodmanOptions) (*PodmanClient, error) {
+func New(opts options.PodmanOptions, agentID string) (*PodmanClient, error) {
+	agentTmp := cruntimes.BaseTempDirName(agentID)
```

Comment:
return the error now?

Reply:
So return it then panic?   Is that really useful?  

Not to mention, how is there never not a current user in the first place?  :-)

Reply:
return it and then the agent will exit, `agent.go` already handles all other errors from constructing dependencies

Reply:
got it.